### ISSUE TITLE
RDLibrary. Edit Audio/Markers Interface Issues

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#732]
Not sure whether this is a feature or an issue...here's what I'm experimenting.

I have instinctively pressed the play button twice, after some time playing, wondering whether the classic Play/Pause functionality often featured in players play/pause button is coded in this player.
However, what I get is that, upon doing the second press on the play button, the vuMeters seem to be like toggled off and the waveform-position green bar halts in place.
Curiously, the music keeps playing just fine though. Moreover, clicking the waveform to move playback to a certain position works too, although the bar keeps static. 

Upon pressing stop, the player successfully 'resets' from this situation.
A new, single, play button press causes the player to work normally.

Regards.

**Comment by ElvishArtisan** [comment]
Unable to reproduce the problem here. What sound setup are you using? (I used HPI, with an ASI6585 card for my tests).

**Comment by alexolivan** [comment]
VM here... I use just rivendell jackd input/outputs... maybe some pesky Jackd issue

I'm planning to install in a physical machine but I have not available at this moment.
Sometimes I think on 'make install' on my laptop, but my experience is very bad with this, since this usually ends up with my Debian install 'windowized', with tons of stuff copied around and left over the filesystem forever.

**Comment by ElvishArtisan** [comment]
Tested with Jack2 on CentOS 7, bound to the 'garden-variety' motherboard audio device on my dev system. Still unable to reproduce the problem.

Question: you have enabled event pausing by ticking the 'Enable Paused Events' checkbox in the 'Miscellaneous Settings' section of RDAdmin->ManageHosts->RDAirPlay, correct?

**Comment by alexolivan** [comment]
> Question: you have enabled event pausing by ticking the 'Enable Paused Events' checkbox in the 'Miscellaneous Settings' section of RDAdmin->ManageHosts->RDAirPlay, correct?

Yes I did.
Just in case, I unticked that and restarted RDAirPLay, and tried again in RDLibrary->EditEvent seeing the same behaviour.
Clicking the play button very repeatedly causes momentaneous vuMeters litUp... it feels very frontend-ish

I'm starting to feel as if some of this UI issues sometimes could merely be caused due to qt5 version mismatches (it seems that sensible changes happened between qt5 versions as they evolve).

**Comment by ElvishArtisan** [comment]
> I'm starting to feel as if some of this UI issues sometimes could merely be caused due to qt5 version mismatches (it seems that sensible changes happened between qt5 versions as they evolve).

Could be. What distro/Qt5 version are you using?

**Comment by alexolivan** [comment]
Hi! Since early September I upgraded my testing VM to the new Debian Stable release (11/ Bullseye)
Qt5 on Bullseye seems to feature a quite (still) recent version, considering the usual lag of featured software on Debian Stable.

```
alex@z800:~$ qmake --version
QMake version 3.1
Using Qt version 5.15.2 in /usr/lib/x86_64-linux-gnu
```

**Comment by ElvishArtisan** [comment]
Yup, that's definitely way newer than what's in RHEL 7 (5.9.7).

Let's spin up Bulleye and see what happens...

**Comment by alexolivan** [comment]
I'm reopening this issue, since, either I'm experiencing an unnoticed issue (till now) or the last fix has screwed up a little the markers edit... I attach an screenshot on what I'm experiencing:

Here you see a cut where there're fade up, down segue and talk markers. But notice that the time grid shows barely no times, everything at 00:00:00 and disabled.

![Screenshot at 2021-11-02 12-13-08](https://user-images.githubusercontent.com/1419896/139836463-41518436-58f8-4ff6-8054-5263c6244af6.png)

Clicking at a marker drag/move triangle on a waveform mark, the talk mark here, just to try to 'shake' the front-end, leads to some changes on the timing grid, here the Fade-Up timing seems to react somehow:

 
![Screenshot at 2021-11-02 12-21-07](https://user-images.githubusercontent.com/1419896/139837927-2cc698ed-fb91-4d66-8227-7c71b6e36e0f.png)

It seems as a kind of Front-end/Qt5 issue reading the mark time values...

**Comment by alexolivan** [comment]
In addition to the issues I found yesterday, I'm wondering whether my experience with the 'Go to Cursor' button is the intended one, in the sense that, it works, as long as the cursor stays visible on the currently shown waveform segment.
When combining a cursor set, followed by an scale change, the 'Go to Cursor' button either does not work, or it resets its position to the very start of the cut...maybe its me, but I find it behaves somehow weird from UX perspective.

Here's a typical use case / procedure I'm finding this:
- I have the full wave form in sight, I play until I reach a point of interest, somewhere int the middle of the track, where, for instance, I want to setup the hook marks.
- 'Go to cursor' button has no usage at this point...the waveform fits the window.
- I click 'Full In', to have full granularity... Cursor disappears as I see full zoom somewhere on the wave form
- I click 'Go to Cursor'
- The waveform is moved at the very start, the cursor position is reset to the very start (losing my reference point).

Regards.

**Comment by alexolivan** [comment]
Yet another issue (I attach screenshot):
For some reason the 'Fade Down Marker' gets greyed-out/disabled even though there's no such mark defined.

![Screenshot at 2021-11-04 11-43-28](https://user-images.githubusercontent.com/1419896/140301003-94878518-3ce6-4275-9e1d-5ccdfa6966a9.png)

In order to re-enable the marker, you need to browse other cart, open editor, and click ok... it's like for some reason, during the operation, some front-end data gets stuck and needs to be reset.

Regards.

**Comment by ElvishArtisan** [comment]
> Yet another issue (I attach screenshot):
For some reason (I can reproduce this consistently only when/if under FULL-IN condition/view) the 'Fade Down Marker' gets greyed-out/disabled even though there's no such mark defined.

Confirmed here.

**Comment by ElvishArtisan** [comment]
> Yet another issue (I attach screenshot):
For some reason (I can reproduce this consistently only when/if under FULL-IN condition/view) the 'Fade Down Marker' gets greyed-out/disabled even though there's no such mark defined.

Fixed in 1bf266a.

However, I have been unable to reproduce the other problems that you've reported. It's conceivable that 1bf266a has also fixed them (or at least changed their behavior). Can you test again?

**Comment by alexolivan** [comment]
I'm no longer able to reproduce the fade down mark issue.... I think that bug is crushed :-)

However, the problem of the go-to-cursor button on full fade in is easy to reproduce... maybe I didn't explain it well:

![Screenshot at 2021-12-30 21-18-41](https://user-images.githubusercontent.com/1419896/147785821-c9c845c4-b739-46ae-8df3-b7c7fbc340fa.png)

1 - Open a cut, and then, without any zooming or operation, just click the waveform to set a cursor position
2 - Click on Full-In button (now, you're at the start of the cut, at full in. No green bar. Cursor position time but states an available position previously set)
3 - Click Go-to-Cursor

Expected behaviour: the waveform is moved, from the beginning of the cut, to the point (wherever it is) in the timeline where the position is set, with the green bar at the cursor position on the timeline.

Actual behaviour: the waveform does not move. The green bar of the cursor indicator appears at the edge of the timeline/waveform as if cursor position would be 00:00:00:00, but cursor position keeps indicating the actual set position.

Cheers!

**Comment by ElvishArtisan** [comment]
Fixed in de6abc0. Please test!

**Comment by alexolivan** [comment]
Yep ... Fixed!

Closing the issue!
